streaming/src/main/scala/org/apache/spark/streaming/util/RecurringTimer.scala

```diff
@@ -99,6 +99,8 @@ class RecurringTimer(clock: Clock, period: Long, callback: (Long) => Unit, name:
         nextTime += period
         logDebug("Callback for " + name + " called at time " + prevTime)
       }
+      clock.waitTillTime(nextTime)
```

Comment:
BTW, are you updating the `prevTime` which is returned by `stop()` as the last time the call was made?
I think there arent unit tests for this, could you add them, so that this behavior is properly captured.